src/common/snippets/docs/snippets_design_guide.md

```diff
@@ -1,301 +1,465 @@
+The `TargetMachine` is a class that provides generator with target-specific information, such as supported instruction sets, vector register size etc. `TargetMachine` also maps the nGraph's `DiscreteTypeInfo` (stored in the `Expression`) to the emitter that actually implements the operation. The mapping is done using the `jitters` map defined in [target_machine.hpp](https://github.com/openvinotoolkit/openvino/blob/master/src/common/snippets/include/snippets/target_machine.hpp). 
+In order for this mechanism to work, every `Snippets'` code generation backend should create emitter implementations derived from the `Emitter` base class defined in [emitter.hpp](https://github.com/openvinotoolkit/openvino/blob/master/src/common/snippets/include/snippets/emitter.hpp). The backend then should create its own target machine class (derived from the common `TargetMachine`) and populate the `jitters` map, see the [cpu_generator.cpp](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/emitters/x64/cpu_generator.cpp#) for an implementation example.
+
+Note that `init_emitters(...)` only initializes the appropriate emitters, but do not actually emit any code. To perform code emission, a `snippets::op::Kernel` operation is constructed (see [generator.cpp](https://github.com/openvinotoolkit/openvino/blob/master/src/common/snippets/src/generator.cpp)), its constructor takes the `IR` with all the initialized emitters as an only input argument. Then a `KernelEmitter` is created from the `Op` using the standard mapping approach. Finally, the `kernel->emit_code({}, {})` command initiates the code emission. Note that the `emit_code(...)` is called only for the `KernelEmitter`, and the emitter is responsible for calling the same method for the rest of the expressions in the `IR` This encapsulation is needed because the `KernelEmitter` performs mapping of the assigned abstract registers to physical registers available on a particular platform. Another important function of the `KernelEmitter` is to calculate input/output data offsets based on dimension indices provided in runtime, and to shift corresponding data-handling registers accordingly. Keep in mind however, that the required functionality of the `KernelEmitter` depends on how the rest of the emitters are implemented (particularly for `Load`/`Store` `Ops`). We've discussed above how the emitters for the `intel_cpu` plugin are implemented (see [jit_snippets_emitters.cpp](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/emitters/x64/jit_snippets_emitters.cpp) for more details), but a different backend might require a different approach depending on hardware specifics.
 
 ## See also
 
  * [OpenVINO™ README](../../../../README.md)
- * [OpenVINO SnippetS](../README.md)
+ * [OpenVINO Snippets](../README.md)
  * [OpenVINO Core Components](../../../README.md)
  * [Developer documentation](../../../../docs/dev/index.md)
+
```

Comment:
Maybe it'd be really great if we clarify some limitations (for example, we support only these precisions, support elementwise operations and some operation only from MHA pattern (and links to the code with this list)), future plans.

What do you think? Seems like we should always update this document but I think that we really should do it before each Release.

Reply:
Yes, I think it's a good idea. We should create a `limitations` subsection in every section (Tokenization, Precision propagation, Generation), and summarize it (just one sentence + link) in a high-level "Present Limitations" section.

Totally agree, we should dedicate more time to documentation development, and update the docs on a regular basis.